### ISSUE TITLE
HTTP updates for Firefox 81

### ISSUE THREAD

**Issue by chrisdavidmills** [mdn/sprints#3720]
## Acceptance criteria

The following bugs are dev-doc-complete, BCD is updated, an entry to Firefox release notes is added, experimental features page is updated, the content has been reviewed as needed.

1. - [ ] [Web Share should be controlled by permission policy](https://bugzilla.mozilla.org/show_bug.cgi?id=1653199) — I think this should be pretty small and easy to do; basically just a simple addition plus BCD on the [Feature-Policy](https://wiki.developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Feature-Policy) page, plus maybe [Feature Policy](https://wiki.developer.mozilla.org/en-US/docs/Web/HTTP/Feature_Policy), and maybe update information about the `allow` attribute?

Also, while you're at it you could maybe do a global change of Feature Policy to Permissions Policy, if that change has come into effect yet.

Estimated total: Anywhere between a couple of hours and a day or two, depending on how much of the above described work is done.

**Comment by hamishwillee** [comment]
Status/Notes for [Web Share should be controlled by permission policy](https://bugzilla.mozilla.org/show_bug.cgi?id=1653199)
- So basically I think this is saying that prior to now web-share could not be controlled, and now it can - added in version 81
- The main page to update would be this one: [Web technology for developers > HTTP > HTTP headers > Feature-Policy > web-share](https://wiki.developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Feature-Policy/web-share) and possibly the parent.
  - This says the feature is experimental. How do I tell if it is not anymore? Or does the fact this appeared here mean it is not?
- [ ] BCD 
  - yes is missing (hasn't populated [into mdn docs here](https://wiki.developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Feature-Policy/web-share#Browser_compatibility) and I can't see it in [http/headers/feature-policy.json](https://github.com/mdn/browser-compat-data/blob/master/http/headers/feature-policy.json).
  - how do I find out the support level on other browsers? Or do I just add this one that I know?
- [ ] maybe do a global change of Feature Policy to Permissions Policy, if that change has come into effect yet.
  - How do you tell? 
  - The [current spec](https://w3c.github.io/webappsec-permissions-policy/#permissions-policy-http-header-field) is a very recent editor's draft.
  - How would I know it is in FF yet to use this rather than Feature-Policy. 
  - When you say global - across all of MDN and in the BCD?
- [ ] I think this should be pretty small and easy to do; basically just a simple addition on the [Feature-Policy](https://wiki.developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Feature-Policy) page
  - It is already there. I think we just concentrate on the specific page.
- [ ] plus maybe [Feature Policy](https://wiki.developer.mozilla.org/en-US/docs/Web/HTTP/Feature_Policy)
   - Maybe just add it to the list in "Granular control over certain features"
- [ ] and maybe update information about the `allow` attribute?
   - What did you have in mind here?
- [ ] Check release note.

**Comment by chrisdavidmills** [comment]
Some answers for you:

> This says the feature is experimental. How do I tell if it is not anymore? Or does the fact this appeared here mean it is not?

Our definition of experimental is "only supported in one browser", and/or "spec is not stable and subject to change". Obviously the former is easier to discern than the latter; generally we stop marking things as experimental when they start being supported in more than one browser; you can remove these banners, in this case. 

> how do I find out the support level on other browsers? Or do I just add this one that I know?

Finding out exhaustive browser support info is tricky. It is better to just fill in what you can find out fairy easily, from Bugzilla, chromestatus.com, webkit.org, and your own testing (`true` is not as good as an exact support version, but it is still much better than `null`), and then the BCD folks can help you fil in the rest of the data. Basically just say what you did, where you got the data from, and what you missed.

Another tip — you can largely infer support versions of other Chromiums from the Chrome support version, as they nearly always just take the upstream without any break in service.

> maybe do a global change of Feature Policy to Permissions Policy, if that change has come into effect yet.
> How do you tell?

I'm not sure; it is just something I read in the bug somewhere. You'd have to ask the engineer for info on this.

> What did you have in mind here?

A lot of feature policy stuff can be set at an indvidual `<iframe>` basis using the `allow` attribute, see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe#Attributes. Although that section points to a different page for all the details. You'll have to follow the rabbit hole and see where it makes sense to include such information ;-)

**Comment by hamishwillee** [comment]
@chrisdavidmills Thanks for the the info. 

Regarding experimental, the general case of Feature Request, and the specific case of web-share. 
- The [bug fix](https://phabricator.services.mozilla.com/D84121) that originates this [shows this page](https://hg.mozilla.org/mozilla-central/file/tip/dom/security/featurepolicy/FeaturePolicyUtils.cpp#l27) which divides the features into supported/experimental and puts web-share in supported. This kind of implies that FeatureRequest is no longer experimental and neither is web-share - but some of the others are.
- Similarly the [BCD](https://github.com/mdn/browser-compat-data/blob/master/http/headers/feature-policy.json) appears to treat FeatureRequest as supported and sub parts as experimental.

So IMO we should move the experimental notes down to the feature level.

With respect to `web-share` though it is not quite so clear where it fits. It is supported by FF, and [experimental on Chromium](https://chromestatus.com/features/6362499966304256). I can't find any other mentions on basic search. My thinking is that we base this on the FF code and base our supported/not supported list on what the developers think. Is that reasonable?

**Comment by chrisdavidmills** [comment]
I think, if in doubt, leave off the experimental banner; it often causes more confusion than it's worth. But yes, I think it'd wise to ask the engineers for clarification. FYI, the best way to get engineers' attention is to check the "Request information from" box at the bottom of the page, and put their usernames in there. That creates a "needinfo", which mails them to ask for help on a bug.

Also FYI, marcos has left Mozilla. You can still try pinging him, but I'm not sure if he'll respond or not at this point.

**Comment by hamishwillee** [comment]
FYI, this is still waiting on the owner to get back to me on exactly where it is supported. Marcos answered and agrees that this is not obvious ("there are questions to answer") from the code.

**Comment by hamishwillee** [comment]
@chrisdavidmills There is an open question on this one - essentially whether the web-share policy only works for domain `self` (as it looks like in the code), and if so, what happens if another domain is set (e.g. src). 
I also asked about when the change to PermissionPolicy is scheduled (more for planning) but that is not as important.

Marcos confirmed there is a question to answer on this, but deferred it to Baku in RequestInfo. I further emailed Baku to no response.

Can you please advise on how to escalate in this kind of case? This is FF81.

**Comment by chrisdavidmills** [comment]
I've reviewed this, made a few small changes to the pages, and this looks pretty good.
